schema.sql

```diff
@@ -1,19 +1,11 @@
-# This file is only used in development environments, and to serve as documentation.
-# It is not used in production in any way.
-
-drop table if exists wp_wporg_gp_translation_events_actions;
-
-create table wp_wporg_gp_translation_events_actions
-(
-    event_id       int(10)     not null comment 'ID of the event',
-    user_id        int(10)     not null comment 'ID of the user who made the action',
-    translation_id int(10)     not null comment 'ID of the translation',
-    action         varchar(16) not null comment 'The action that the user made (create, reject, etc)',
-    locale         varchar(10) not null comment 'Locale of the translation',
-    # Make sure that for a given event and translation, the user cannot do multiple actions.
-    primary key (event_id, user_id, translation_id)
-)
-    comment 'Tracks translation actions that happened during a translation event';
-
-create index wp_wporg_gp_translation_events_actions_event_id_locale_index
-    on wp_wporg_gp_translation_events_actions (event_id, locale);
+CREATE TABLE `translate_event_actions` (
+  `translate_event_actions_id` int(11) unsigned NOT NULL AUTO_INCREMENT,
+  `event_id` int(10) NOT NULL COMMENT 'Post_ID of the translation_event post in the wp_posts table',
+  `original_id` int(10) NOT NULL COMMENT 'ID of the translation',
+  `user_id` int(10) NOT NULL COMMENT 'ID of the user who made the action',
+  `action` enum('approve','create','reject','request_changes') NOT NULL COMMENT 'The action that the user made (create, reject, etc)',
+  `locale` varchar(10) CHARACTER SET latin1 COLLATE latin1_swedish_ci NOT NULL COMMENT 'Locale of the translation',
+  `happened_at` datetime NOT NULL COMMENT 'When the action happened, in UTC',
+  PRIMARY KEY (`translate_event_actions_id`),
+  UNIQUE KEY `event_per_translated_original_per_user` (`event_id`,`locale`,`original_id`,`user_id`)
```

Comment:
A user can add several actions with the same item (locale and original string) that should be registered. E.g., a validator that rejects a translation and then adds a new translation for the same locale and original string. With the current key, it will count as one action, but it should be 2 actions. 

```suggestion
  UNIQUE KEY `event_per_translated_original_per_user` (`event_id`,`locale`,`original_id`,`user_id`, `action`)
```

Reply:
We had actually agreed that we don't want to count this as two actions so that we don't encourage behavior where someone would create a translation as waiting and then approve it. I'd say rejecting and fixing counts as review an this is how it's implemented.

Reply:
Now that were tracking actions on the original instead of the translation, I'm wondering if we need to reconsider the "one action per user per locale" thing. Because indeed like @amieiro mentions, new translations or review actions on the same original by the same user won't count towards stats.

For example, if there are multiple waiting translations for the same original, and a reviewer approves one but rejects the other, only one of those review counts towards stats, while the reviewer did do the work twice.

It's a subtle thing, probably an edge case, but if we would want to address it, one simple way could be to also add the `translation_id` to the table, and include it in the unique key instead of `original_id`.

Reply:
As soon as one suggestion is approved the other suggestions become `old`, so I see it as the review work is done only once.

Reply:
After re-reading the reasoning for using `original_id` instead of `translation_id`, I guess my suggestion above of also adding `translation_id` doesn't really make sense as it would defeat the purpose.

I guess it's a tradeoff between two edge cases: preventing users from gaming stats, or accurately counting actions on the same translation.

Reply:
I see it this way: you can get one credit for working on a translation of an original, either translate it, or review it.

Reply:
> As soon as one suggestion is approved the other suggestions become old, so I see it as the review work is done only once.

That makes sense, I wasn't aware of this.